### PR TITLE
Phase2-hgx231 Make the new masking a possibility

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Read in an additional parameter to make the new masking a possibility

#### PR validation:

Use the standard runTheMatrix.py

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 1]
```diff
@@ -1373,11 +1374,26 @@ void HGCalGeomParameters::loadWaferHexagon8(HGCalParameters& php) {
...
-                                        << " corners First " << corner.second;
+          edm::LogVerbatim("HGCalGeom") << "Layer " << lay << " u|v " << u << ":" << v << " with corner "
+                                        << corner.first << ":" << corner.second;
 #endif
           int wl = HGCalWaferIndex::waferIndex(lay, u, v);
-          waferTypes[wl] = std::make_pair(corner.first, corner.second);
+          if (php.waferMaskMode_ > 0) {
+            std::pair<int, int> corner0 = HGCalWaferMask::getTypeMode(
+                xpos, ypos, r, R, php.rMinLayHex_[i], php.rMaxLayHex_[i], N, php.waferMaskMode_);
+            waferTypes[wl] = std::make_pair(corner0.first, corner0.second);
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: This seems redundant. `corner0` is already a pair of ints, so why do you need to call `make_pair()` on its elements?

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 2]
```diff
@@ -1373,11 +1374,26 @@ void HGCalGeomParameters::loadWaferHexagon8(HGCalParameters& php) {
...
+#ifdef EDM_ML_DEBUG
+            edm::LogVerbatim("HGCalGeom")
+                << "Layer " << lay << " u|v " << u << ":" << v << " with corner " << corner.first << ":"
+                << corner.second << " croner0 " << corner0.first << ":" << corner0.second;
+#endif
+          } else {
+            waferTypes[wl] = std::make_pair(corner.first, corner.second);
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: same question, `corner` is also a pair

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
please test workflow 23434.1001

**Comment by silviodonato** [comment]
+1

**Pull request merged**